sdk/rust/src/contract/contract_function_selector.rs

```diff
@@ -0,0 +1,197 @@
+use sha3::{
+    Digest,
+    Keccak256,
+};
+
+#[derive(Debug, Clone)]
+pub struct ContractFunctionSelector {
+    digest: Option<Keccak256>,
+    needs_comma: bool,
+    finished_bytes: Option<[u8; 4]>,
+}
+
+impl From<[u8; 4]> for ContractFunctionSelector {
+    fn from(value: [u8; 4]) -> Self {
+        Self { digest: None, needs_comma: false, finished_bytes: Some(value) }
+    }
+}
+
+impl ContractFunctionSelector {
+    #[allow(dead_code)]
+    pub fn new(func_name: &str) -> Self {
+        let mut digest = <Keccak256 as Digest>::new_with_prefix(func_name.as_bytes());
+        digest.update("(".as_bytes());
```

Comment:
```suggestion
        digest.update(b"(");
```
(and so on for the rest)